nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-value-providers/src/main/java/org/apache/nifi/stateless/parameter/aws/SecretsManagerParameterValueProvider.java

```diff
@@ -0,0 +1,208 @@
+ *      nifi.stateless.parameter.provider.AWSSecretsManager.type=org.apache.nifi.stateless.parameter.aws.SecretsManagerParameterValueProvider
+ *      nifi.stateless.parameter.provider.AWSSecretsManager.properties.aws-configuration-file=./conf/bootstrap-aws.conf
+ * </code>
+ */
+public class SecretsManagerParameterValueProvider extends AbstractParameterValueProvider implements ParameterValueProvider {
+    private static final String QUALIFIED_SECRET_FORMAT = "%s/%s";
+    private static final String ACCESS_KEY_PROPS_NAME = "aws.access.key.id";
+    private static final String SECRET_KEY_PROPS_NAME = "aws.secret.access.key";
+    private static final String REGION_KEY_PROPS_NAME = "aws.region";
+
+    public static final PropertyDescriptor AWS_CONFIG_FILE = new PropertyDescriptor.Builder()
+            .displayName("AWS Configuration File")
+            .name("aws-configuration-file")
+            .required(false)
+            .defaultValue("./conf/bootstrap-aws.conf")
+            .description("Location of the bootstrap-aws.conf file that configures the AWS credentials.  If not provided, the default AWS credentials will be used.")
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    private final ObjectReader objectReader = new ObjectMapper().reader();
+
+    private final Set<String> supportedParameterNames = new HashSet<>();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private AWSSecretsManager secretsManager;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    protected void init(final ParameterValueProviderInitializationContext context) {
+        super.init(context);
+
+        this.descriptors = Collections.unmodifiableList(Arrays.asList(AWS_CONFIG_FILE));
+
+        final String awsBootstrapConfFilename = context.getProperty(AWS_CONFIG_FILE).getValue();
+        try {
+            this.secretsManager = this.configureClient(awsBootstrapConfFilename);
+        } catch (final IOException e) {
+            throw new IllegalStateException("Could not configure AWS Secrets Manager Client", e);
+        }
+
+        cacheSupportedParameterNames();
+    }
+
+    private void cacheSupportedParameterNames() {
+        supportedParameterNames.clear();
+        final ListSecretsResult listSecretsResult = secretsManager.listSecrets(new ListSecretsRequest());
+        if (listSecretsResult != null) {
+            for (final SecretListEntry entry : listSecretsResult.getSecretList()) {
+                supportedParameterNames.add(entry.getName());
+            }
+        }
+    }
+
+    @Override
+    public boolean isParameterDefined(final String contextName, final String parameterName) {
+        return supportedParameterNames.contains(getSecretName(contextName, parameterName));
+    }
+
+    @Override
+    public String getParameterValue(final String contextName, final String parameterName) {
```

Comment:
Minor suggestion, what do you think about moving these `public` methods up in the class before other `protected` and `private` methods?